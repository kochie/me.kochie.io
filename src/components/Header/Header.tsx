import * as React from "react";
import { Chart } from "chart.js";
import Head from "next/head";

export default class Header extends React.Component {
  canvas: React.RefObject<HTMLCanvasElement>;
  chart?: Chart;

  constructor(props: Readonly<{}>) {
    super(props);
    this.canvas = React.createRef();
  }

  componentDidMount() {
    if (this.canvas.current === null) {
      return;
    }
    const ctx = this.canvas.current.getContext("2d");
    if (ctx === null) {
      return;
    }
    const options = {
      type: "line",
      data: {
        labels: ["", "", "", "", "", ""],
        datasets: [
          {
            label: "# of Votes",
            data: [12, 1, 3, 5, 10, 4],
            borderWidth: 1,
            pointRadius: 0
          }
        ]
      },
      options: {
        animation: { duration: 2000, easing: "linear" },
        gridLines: {
          drawTicks: false,
          lineWidth: false,
          drawBorder: true
        },
        ticks: {
          display: false,
          fontColor: "#666"
        },
        legend: {
          display: false
        },
        maintainAspectRatio: false,
        layout: {
          padding: {
            bottom: 0
          }
        },
        scales: {
          xAxes: [
            {
              display: false
            }
          ],
          yAxes: [
            {
              display: false
            }
          ]
        }
      }
    };
    this.chart = new Chart(ctx, options);

    setInterval(() => this.updateChart(), 5000);
  }

  updateChart() {
    console.log("updated");
    if (!this.chart || !this.chart.data.datasets) {
      return;
    }
    const list = this.chart.data.datasets[0].data;
    if (typeof list === "undefined") {
      return;
    }
    list.splice(0, 1);
    list.push(Math.random() * 10);
    this.chart.update();
  }

  render() {
    return (
      <>
        <Head>
          <link rel="stylesheet" href="/static/styles/header.css" />
        </Head>
        <div className="header">
          <canvas className="backchart" ref={this.canvas} />
          <div className="gradient" />
          <span className="title-firstname">ROBERT</span>
          <br />
          <span className="title-lastname">KOCH</span>
        </div>
      </>
    );
  }
}
