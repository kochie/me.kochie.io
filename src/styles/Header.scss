@import url(https://fonts.googleapis.com/css?family=Bowlby+One+SC|Bungee);

.header {
  background-color: lightgray;
  grid-area: header;
  position: relative;
  max-height: 150px;
  height: 150px;
}

.title-firstname {
  position: absolute;
  top: 0px;
  line-height: 0.79;
  vertical-align: baseline;
  font-family: 'Bowlby One SC', cursive;
  font-size: 100px;
  font-style: normal;
}

.title-lastname {
  position: absolute;
  bottom: 0px;
  line-height: 0.73;
  vertical-align: baseline;
  font-family: 'Bowlby One SC', cursive;
  font-size: 100px;
  font-style: normal;
}

.backchart{
  height: 150px
}

.gradient {
  height: 100%;
  width: 100%;
  position: absolute;
  top: 0;
  background: linear-gradient(
    to right, rgba(211,211,211,1) 0%,
    rgba(211,211,211,1) 31%
    ,rgba(211,211,211,0) 67%,
    rgba(211,211,211,0) 100%
  );
}