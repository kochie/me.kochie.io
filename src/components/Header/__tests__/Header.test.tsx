import * as renderer from "react-test-renderer";
import * as React from "react";

import Header from "../Header";

it("header renders correctly", () => {
  const tree = renderer.create(<Header />).toJSON();
  expect(tree).toMatchSnapshot();
});
