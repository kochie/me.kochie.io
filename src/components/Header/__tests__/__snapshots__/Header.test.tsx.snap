// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`header renders correctly 1`] = `
<div
  className="header"
>
  <canvas
    className="backchart"
  />
  <div
    className="gradient"
  />
  <span
    className="title-firstname"
  >
    ROBERT
  </span>
  <br />
  <span
    className="title-lastname"
  >
    KOCH
  </span>
</div>
`;
